File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
      
   
        public ListNode deleteDuplicates(ListNode head) {
        
    if(head==null)
    return null ;
ListNode temp = head.next ;
int val = head.val ;
ListNode prev = head ;
    while(temp!=null){

        if(temp.val==val){
                prev.next = temp.next ;
              
                val = temp.val ;
                temp = temp.next; 
        }else{
        val = temp.val ;
        prev = temp ;
        temp = temp.next; 
        }
    }

return head ;
}

}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
#define Node struct ListNode*
struct ListNode* deleteDuplicates(struct ListNode* head) {
    Node temp = head ;
    while(temp!=NULL){
        Node temp2 = temp->next ; 
        while(temp2!=NULL&&temp->val==temp2->val){
            temp2 = temp2->next ;
        }
        temp->next = temp2 ;
        temp = temp->next;
    }
    return head;
}